app/src/main/java/org/oppia/android/app/onboarding/OnboardingFragmentPresenter.kt

```diff
@@ -36,13 +80,198 @@ class OnboardingFragmentPresenter @Inject constructor(
         appLanguageResourceHandler.getStringInLocale(R.string.app_name)
       )
 
+      onboardingAppLanguageViewModel.supportedAppLanguagesList.observe(
+        fragment,
+        { languagesList ->
+          val adapter = ArrayAdapter(
+            fragment.requireContext(),
+            R.layout.onboarding_language_dropdown_item,
+            R.id.onboarding_language_text_view,
+            languagesList.map { appLanguageResourceHandler.computeLocalizedDisplayName(it) }
+          )
+          onboardingLanguageDropdown.setAdapter(adapter)
+        }
+      )
+
+      onboardingAppLanguageViewModel.languageSelectionLiveData.observe(
+        fragment,
+        { language ->
+          selectedLanguage = language
+          onboardingLanguageDropdown.setText(
+            appLanguageResourceHandler.computeLocalizedDisplayName(
+              language
+            ),
+            false
+          )
+        }
+      )
+
+      onboardingLanguageDropdown.apply {
+        setRawInputType(EditorInfo.TYPE_NULL)
+
+        onItemClickListener =
+          AdapterView.OnItemClickListener { _, _, position, _ ->
+            adapter.getItem(position).let { selectedItem ->
+              selectedItem?.let {
+                val localizedNameMap = OppiaLanguage.values().associateBy { oppiaLanguage ->
```

Comment:
I'm not sure that this is the correct way to handle this. The enum includes an unspecified constant, plus it may contain languages not yet ready for full release in the app (which is why we have language configurations).

Might ``getSupportedAppLanguages`` be a better fit here? It should already do the necessary work for filtering valid languages:

https://github.com/oppia/oppia-android/blob/4730edb0263135e88dab08ab0f6dc724a1585a67/domain/src/main/java/org/oppia/android/domain/translation/TranslationController.kt#L114

I'm not sure if the returned list's contents and order are stable--you'll need to double check that.

Reply:
How does the mapping work, exactly? This is iterating through the enum, but if some of the languages in the enum are excluded from the list returned by ``getSupportedAppLanguages`` won't the selected index be potentially wrong (and thus set the wrong language)?

In fact, I think this would be broken on production builds (e.g. ``//:oppia_beta``). ``oppia_dev`` uses the 'all supported languages' config file, but the production one uses this:

https://github.com/oppia/oppia-android/blob/dfb9a301280b9a46526cb2f5ca6329532fec6bf0/config/src/java/org/oppia/android/config/productionlanguages/supported_languages.textproto#L1-L105

When compared with the language proto:

https://github.com/oppia/oppia-android/blob/dfb9a301280b9a46526cb2f5ca6329532fec6bf0/model/src/main/proto/languages.proto#L9-L69

Hindi isn't supported, so presumably there would be a selection issue on production builds.

Or, am I misunderstanding still?

Reply:
You're right, since I'm relying on the position from the dropdown to fetch the selected language, the index of the dropdown item could no longer match the index of the language in the enum. This mismatch could cause the selected language to be wrong. So I have refactored the code to rely on the supported app language list, and not the OppiaLanguage enum. I have also tested with the beta build.

[device-2024-10-01-144818.webm](https://github.com/user-attachments/assets/da0fe694-2e74-4c1e-ad86-6f96aa9022f6)

Reply:
Ah--thanks for digging into this more! Latest solution LGTM.